src/authn/index.js

```diff
@@ -602,6 +607,26 @@ async function verifyToken(token, use, client = COGNITO_CLIENT_ID) {
     throw new JWTClaimValidationFailed(`unexpected "token_use" claim value: ${claimedUse}`, "token_use", "check_failed");
   }
 
+  /* Verify the token was issued at (iat) a time more recent than our staleness
+   * horizon for the user.
+   */
+  const username = claims[{id: "cognito:username", access: "username"}[claimedUse]];
```

Comment:
Nod. This was something I checked, but I should maybe note it in the code.

Usernames can be _reused_ if someone's account is deleted and another account is created. I think this is harmless for these keys, but worth noting in general. There is a `sub` (subject) UUID which is totally immutable and unique in time which is the "most proper" thing to use here, and maybe we should do more with that in general (though it would make these keys no longer self-descriptive).

It occurs to me I should also check if username is stable even for third-party identities, which I do think we should enable sooner than later (but requires some more work on our end yet).

Reply:
Using the `sub` UUID sounds like a good alternative. Is there any reason the keys need to be self-descriptive?

Reply:
I looked into this again right now and re-realized why `username` is a good choice for this but `sub` isn't. It's not as obvious here because we only call `userStaleBefore()` when we have `sub` available and never call `markUserStaleBefore()`. But in future work, we'll absolutely be needing to call `markUserStaleBefore()` in contexts where we don't have the `sub` of the user we're marking stale, just the username, and I don't think we don't want to impose extra lookups of username → sub on those calls.

For example, one of those future contexts is a group owner changing the membership role of another user. This is an operation that will naturally occur by username at first. Maybe later we'll support searching by username or email and at that point having paid the price of a Cognito ListUsers call already we can switch this key from username → sub.

Reply:
In my head I was thinking through groups management via a web UI, so I thought we would have `sub` available since we'd have to look up all of the group's users to populate a dropdown list. However, thinking through this from the CLI perspective, makes sense that they would not have `sub` available.

Reply:
Ah, nod. I was thinking the web API for this would be more ergonomic if it took usernames, even if one client (the web UI) had easy access to `sub`. Of course, we could make it take `sub` instead (or in addition to!) and have an endpoint to help do lookups, but ¯\_(ツ)_/¯ I think it's fine to not worry about this for now. I don't think it will be hard to change in the future.